fe/fe-core/src/main/java/com/starrocks/planner/RangeDistributionPruner.java

```diff
@@ -0,0 +1,171 @@
+
+            // 1. IN predicate literals (point queries)
+            List<LiteralExpr> inPredicateLiterals = filter.getInPredicateLiterals();
+            if (inPredicateLiterals != null && !inPredicateLiterals.isEmpty()) {
+                for (LiteralExpr expr : inPredicateLiterals) {
+                    Variant v = Variant.of(column.getType(), expr.getStringValue());
+                    lowerValues.add(v);
+                    upperValues.add(v);
+                }
+            }
+
+            // 2. Range bounds (>, >=, <, <=, BETWEEN, =)
+            Variant colMin = null;
+            Variant colMax = null;
+            LiteralExpr lowerBound = filter.getLowerBound();
+            if (lowerBound != null) {
+                colMin = Variant.of(column.getType(), lowerBound.getStringValue());
+            }
+
+            LiteralExpr upperBound = filter.getUpperBound();
+            if (upperBound != null) {
+                colMax = Variant.of(column.getType(), upperBound.getStringValue());
+            }
+
+            // 3. Fill unbounded sides with type-wide min/max
+            boolean hasRangeBound = colMin != null || colMax != null;
+            if (colMin == null) {
+                colMin = Variant.minVariant(column.getType());
+            }
+            if (colMax == null) {
+                colMax = Variant.maxVariant(column.getType());
+            }
+            if (hasRangeBound) {
+                lowerValues.add(colMin);
+                upperValues.add(colMax);
+            } else if (lowerValues.isEmpty() && upperValues.isEmpty()) {
+                // no any predicate on this column => prune range by the prefix keys
+                break;
+            }
+
+            complex = complex * lowerValues.size();
+            if (complex > Config.max_distribution_pruner_recursion_depth) {
+                // prevent combinatorial explosion, early stop
+                break;
+            }
+            lowerValuesLists.add(lowerValues);
+            upperValuesLists.add(upperValues);
+        }
+
+        // fast path: no range pruning is needed
+        if (lowerValuesLists.isEmpty() && upperValuesLists.isEmpty()) {
+            return tabletInOrder.stream().map(Tablet::getId).collect(Collectors.toSet());
+        }
+
+        List<Variant> currentLowerValue = new ArrayList<>();
+        List<Variant> currentUpperValue = new ArrayList<>();
+        return prune(0, lowerValuesLists, upperValuesLists, currentLowerValue, currentUpperValue);
+    }
+
+    private Collection<Long> prune(int columnIdx, List<List<Variant>> lowerValuesLists,
+                                   List<List<Variant>> upperValuesLists,
+                                   List<Variant> currentLowerValue,
+                                   List<Variant> currentUpperValue) {
+        // lowerValuesLists / upperValuesLists size may less than rangeDistributionColumns size
+        if (columnIdx == lowerValuesLists.size()) {
```

Comment:
Pruning can still be performed when the partition filter includes a prefix of the range columns.

Reply:
It has already support the prefix column prune. In `public Collection<Long> prune()`, it will save the prefix column predicate and still perform DFS to search all predicate combination for the prefixes.